game/theater/iadsnetwork/iadsnetwork.py

```diff
@@ -268,3 +313,24 @@ def initialize_network_from_range(self) -> None:
                         <= node.group.iads_role.connection_range.meters
                     ):
                         node.add_connection_for_tgo(nearby_go)
+
+    def _is_friendly(self, node: IadsNetworkNode, tgo: TheaterGroundObject) -> bool:
+        node_friendly = node.group.ground_object.is_friendly(True)
+        tgo_friendly = tgo.is_friendly(True)
+        return node_friendly == tgo_friendly
+
+    def _update_network(
+        self, tgo: TheaterGroundObject, events: GameUpdateEvents
+    ) -> None:
+        if tgo.is_dead:
+            return
+        # Reversing the idea of _connect_to_network...
```

Comment:
Just to be clear, _connect_to_network was renamed _make_advanced_connections_by_range.

I'll adjust it or perhaps remove it in part 4, otherwise it'll mess up the commit history...